chore(deps): replace deprecated lib to alternative

yamatatsu opened the pull request:
<!--- 
Instructions:
1. Make sure you follow our Contributing Guidelines: https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/CONTRIBUTING.md
2. Please follow the template, and do not remove any section/item. If something is not applicable leave it empty, but leave it in the PR. 
3. -->

## Description of your changes

<!---
Include here a summary of the change.

Please include also relevant motivation and context.

Add any applicable code snippets, links, screenshots, or other resources
that can help us verify your changes.
-->

This PR remove deprecated library `@aws-sdk/util-base64-node` replacing to `@smithy/util-base64` that is official alternative library.
See, https://www.npmjs.com/package/@aws-sdk/util-base64-node.

### Related issues, RFCs

<!--- 
Add here the number (i.e. #42) to the Github Issue or RFC that is related to this PR.

Don't include any other text, otherwise the Github Issue will not be detected.

Note: If no issue is present the PR might get blocked and not be reviewed.
-->
**Issue number: #2246** 

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [ ] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

dreamorosi commented:
Hi, thank you for taking the time to open this PR.

As I mentioned in the linked issue, we would like to go in a different direction and instead adopt the `fromBase64` utility that we recently added to the commons package.

If you'd like to continue working on this and see the PR merged I'd kindly ask you to make changes in that regard and remove the `@smithy` dependency.

Apologies if this might cause you some rework, to avoid this type of occurrence in the future I'd advise to wait for a feedback from a maintainer in the issue before opening the PR, that way we'll be aligned from the beginning and we have a better chance to merge the PR with minor adjustments or as is.

dreamorosi commented:
Hi @yamatatsu, are you still interested in continuing this PR? 

If not there's no problem at all and we'll take on the task at some point in the near future.

Please let me know, or we'll close the PR in a week.

yamatatsu commented:
@dreamorosi 
Thank you for feedback. And so sorry for late response. 🙇🏻 
I've addressed your comments.
In my thinking, we can depend on `@smithy/xxx` only as devDependencies. Or we should implement `toBase64()` like as `fromBase64()`? WDYT? 🤔

dreamorosi commented:
Hi @yamatatsu, no problem and thanks for coming back to this.

I agree with you, we can depend on it as a `devDependency` for now since we use `toBase64()` only in the tests.

Thanks!

dreamorosi commented on packages/parameters/src/base/transformValue.ts:
Line 70 needs to be changed to: `return new TextDecoder('utf-8').decode(fromBase64(value, 'base64'));`, the new implementation requires to specify the encoding.

Once that's done the tests will pass.

yamatatsu replied:
Oh... Sorry that I missed that the tests failed.
Thank you for debugging it! 🙇🏻

dreamorosi reviewed (changes requested):
I have left one comment to fix the implementation and make the tests passing, once that's addressed we can merge this!

dreamorosi reviewed (approved):
Thanks a lot for working on this PR and following through with all the feedback!

Congrats 🎉
